pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +657,816 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
+        verboseprint(f'Index file generated at {output_csv_path}')
+        write_results_to_csv(all_results, args, output_csv_path)
+
+    # To instead receive a list of available information available within a label or set
+    # of labels, you may use --output-headers-file. This will take all of the keys of
+    # the label_results dictionary and place them in the output file, instead of the
+    # index file.
+    if output_txt_path:
+        xpaths = []
+        for label in all_results:
+            for values in label.values():
+                for xpath in values.keys():
+                    if xpath not in xpaths:
+                        xpaths.append(xpath)
+
+        # The file is now written and placed in a given location. If cleaned header
+        # field names are requested, they are processed here before being written in.
+        with open(output_txt_path, 'w') as output_fp:
+            for item in xpaths:
+                if args.clean_header_field_names:
+                    verboseprint(
+                        '--clean-header-field-names active. Headers reformatted.')
+                    item = item.replace(
+                        ':', '_').replace('/', '__').replace('<', '_').replace('>', '')
+                output_fp.write("%s\n" % item)
+        verboseprint(f'XPath headers file generated at {output_txt_path}')
+
+    # Generates the label for this index file, if --generate-label is used.
+
+    if args.generate_label:
+        index_file = output_csv_path
+
+        # The template label file and the default_values.yaml file are initialized.
+        module_dir = Path(__file__).resolve().parent
+        yaml_file = module_dir / 'default_values.yaml'
+        tempfile = str(module_dir / 'index_label_template_pds.xml')
+        template = ps.PdsTemplate(tempfile)
+
+        # In this case, this filename is the filename of the index file previously
+        # generated.
+        filename = str(Path(index_file).stem)
+
+        header_info = []
+        sniffer = csv.Sniffer()
+
+        # The index file is opened and read for the contents of the headers. The delimiter
+        # is also found for later reference.
+        with open(index_file, 'r', encoding='utf-8') as index_fp:
+            full_header = index_fp.readline()
+            full_header_length = len(full_header)
+            try:
+                sample_data = index_fp.read(5000)
+                delimiter = sniffer.sniff(sample_data).delimiter
+                index_fp.seek(0)  # Reset file pointer to the beginning
+            except csv.Error:
+                print(f'Index file {index_file} is not a CSV or tab-separated file.')
+                sys.exit()
+            reader = csv.reader(index_fp, delimiter=delimiter)
+            headers = next(reader)
+
+            offset = 0
+            field_number = 0
+            jump = len(delimiter)
+            field_location = 1
+            maximum_field_lengths = max_field_lengths(index_file)
```

Comment:
The way to avoid this awkward use of max vs. maximum is to make the function have a verb in it: `compute_max_field_lengths`.

Reply:
I also added a docstring.